### PR TITLE
[6.5.x] RHBPMS-4677 - Retrieving Form Data with Kie Server Rest API does not show subforms data in BPM Suite 6.4.x

### PR DESCRIPTION
**Pull request by tsurdilo**
this pr pulls in 	JBPM-5973 fix into 6.5.x and adds subform test. api changes for 6.5.x from the master commits were made.

@mswiderski sorry about the 2 subform test commits - was tired and forgot to amend. it can be squashed if merged.

### CODE REVIEW

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/FormResource.java [hunk 1]
```diff
@@ -102,7 +104,9 @@ public Response getProcessForm(@javax.ws.rs.core.Context HttpHeaders headers,
     @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
     public Response getTaskForm(@javax.ws.rs.core.Context HttpHeaders headers,
             @PathParam(CONTAINER_ID) String containerId, @PathParam(TASK_INSTANCE_ID) Long taskId,
-            @QueryParam("lang") @DefaultValue("en") String language, @QueryParam("filter") boolean filter) {
+            @QueryParam("lang") @DefaultValue("en") String language, @QueryParam("filter") boolean filter,
+            @QueryParam("type") @DefaultValue("ANY") String formType, @QueryParam("marshallContent") @DefaultValue("true") boolean marshallContent ) {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/FormResource.java]: don't think 6.5.x supports formType as it only works with .form forms

File: kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/FormResource.java [hunk 2]
```diff
@@ -71,8 +71,10 @@ public FormResource(FormServiceBase formServiceBase, KieServerRegistry context)
     @Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
     public Response getProcessForm(@javax.ws.rs.core.Context HttpHeaders headers,
             @PathParam(CONTAINER_ID) String containerId, @PathParam(PROCESS_ID) String processId,
-            @QueryParam("lang") @DefaultValue("en") String language, @QueryParam("filter") boolean filter) {
+            @QueryParam("lang") @DefaultValue("en") String language, @QueryParam("filter") boolean filter,
+            @QueryParam("type") @DefaultValue("ANY") String formType, @QueryParam("marshallContent") @DefaultValue("true") boolean marshallContent) {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-ui/src/main/java/org/kie/server/remote/rest/jbpm/ui/FormResource.java]: similar here for the formType query param, not supported in 6.5.x

**Comment by tsurdilo** [comment]
@mswiderski amended - thanks!

**Comment by tsurdilo** [comment]
jenkins please retest this

**Comment by tsurdilo** [comment]
jenkins fail is unrelated to the pr

**Comment by tsurdilo** [comment]
@mswiderski tests amended to make sure both filtered and unfiltered forms pass. ready to be merged if approved. thanks.

**Pull request merged**